javascripts/2048.js

```diff
@@ -1,32 +1,347 @@
+        rows[i][j+1].remove();
+    };
+
+  }
+}
+}
+
+
+function moveRowLeft(row) {
+  // for(var i = 0; i < row.length; i++) {
+  //   row[i].setAttribute("data-col", ("c" + i));
+  //   }
+  for (var i = 0; i < row.length; i++) {
+  for(var j = 0; j < row[i].length; j++) {
+    row[i][j].setAttribute("data-col", ("c" + j));
+    }
+  }
+}
+
+
+function collideTilesRight(rows){
+  for (var i = 0; i < rows.length; i++) {
+
+  for (var j = 0; j < rows[i].length -1; j++){
+   var leftTileValue = rows[i][j].getAttribute("data-val");
+   var nextTileValue = rows[i][j+1].getAttribute("data-val");
+      if (leftTileValue == nextTileValue){
+        var newTileValue = nextTileValue * 2;
+          rows[i][j+1].setAttribute("data-val", newTileValue);
+          rows[i][j+1].textContent = newTileValue;
+          rows[i][j].remove();
+    };
+
+  }
+}
+}
+
+function moveRowRight(row) {
+  for (var i = 0; i < row.length; i++) {
+    for(var j = 0; j < row[i].length; j++) {
+    row[i][j].setAttribute("data-col", ("c" + (j + 4 - row[i].length)));
+    }
+  }
+}
+
+function makeCols(tile) {
+  var col0 = [];
+  var col1 = [];
+  var col2 = [];
+  var col3 = [];
+
+  for (var i = 0; i < tile.length; i++) {
+      if (tile[i].getAttribute("data-col") == "c0") {
+        col0.push(tile[i]);
+      } else if (tile[i].getAttribute("data-col") == "c1") {
+        col1.push(tile[i]);
+      } else if (tile[i].getAttribute("data-col") == "c2") {
+        col2.push(tile[i]);
+      } else if (tile[i].getAttribute("data-col") == "c3") {
+        col3.push(tile[i]);
+      }
+    }
+
+  var cols = [col0, col1, col2, col3];
+
+  return cols;
+}
+
+function moveColUp(col) {
+  for (var i = 0; i < col.length; i++) {
+    for(var j = 0; j < col[i].length; j++) {
+      col[i][j].setAttribute("data-row", ("r" + j));
+    }
+  }
+}
+
+function collideTilesUp(cols){
+
+  for (var i = 0; i < cols.length; i++) {
+    for (var j = 0; j< cols[i].length-1; j++){
+      var upTileValue = cols[i][j].getAttribute("data-val");
+      var nextTileValue = cols[i][j+1].getAttribute("data-val");
+      if (upTileValue == nextTileValue){
+        var newTileValue = upTileValue * 2;
+        cols[i][j].setAttribute("data-val", newTileValue);
+        cols[i][j].textContent = newTileValue;
+        cols[i][j+1].remove();
+    };
+
+  }
+}
+}
+
+function moveColDown(col) {
+  for (var i = 0; i < col.length; i++) {
+    for(var j = 0; j < col[i].length; j++) {
+      col[i][j].setAttribute("data-row",("r" + (j + 4 - col[i].length)));
+    }
+  }
+}
+
+function collideTilesDown(cols){
+  for (var i = 0; i < cols.length; i++) {
+
+  for (var j = 0; j < cols[i].length -1; j++){
+   var downTileValue = cols[i][j].getAttribute("data-val");
+   var nextTileValue = cols[i][j+1].getAttribute("data-val");
+      if (downTileValue == nextTileValue){
+        var newTileValue = downTileValue * 2;
+          cols[i][j].setAttribute("data-val", newTileValue);
```

Comment:
These are not inside a block, so should be the same indent level as the var above.